src/video.rs

```diff
@@ -105,68 +118,65 @@ pub fn compare_videos(
-            }
-            (8, _) => {
-                let src_frame = source.read_video_frame::<u8>();
-                let dst_frame = distorted.read_video_frame::<u16>();
-                if let (Some(src_frame), Some(dst_frame)) = (src_frame, dst_frame) {
-                    let src = Yuv::new(src_frame, src_config).unwrap();
-                    let dst = Yuv::new(dst_frame, dst_config).unwrap();
-                    (Rgb::try_from(src).unwrap(), Rgb::try_from(dst).unwrap())
-                } else {
-                    break;
-                }
-            }
-            (_, 8) => {
-                let src_frame = source.read_video_frame::<u16>();
-                let dst_frame = distorted.read_video_frame::<u8>();
-                if let (Some(src_frame), Some(dst_frame)) = (src_frame, dst_frame) {
-                    let src = Yuv::new(src_frame, src_config).unwrap();
-                    let dst = Yuv::new(dst_frame, dst_config).unwrap();
-                    (Rgb::try_from(src).unwrap(), Rgb::try_from(dst).unwrap())
-                } else {
-                    break;
-                }
-            }
-            (_, _) => {
-                let src_frame = source.read_video_frame::<u16>();
-                let dst_frame = distorted.read_video_frame::<u16>();
-                if let (Some(src_frame), Some(dst_frame)) = (src_frame, dst_frame) {
-                    let src = Yuv::new(src_frame, src_config).unwrap();
-                    let dst = Yuv::new(dst_frame, dst_config).unwrap();
-                    (Rgb::try_from(src).unwrap(), Rgb::try_from(dst).unwrap())
+    let (result_tx, result_rx) = mpsc::channel();
+    let src_bd = source.get_bit_depth();
+    let dst_bd = distorted.get_bit_depth();
+
+    let decoders = Arc::new(Mutex::new((source, distorted)));
+    for _ in 0..threads {
+        let decoders = Arc::clone(&decoders);
+        let result_tx = result_tx.clone();
+
+        std::thread::spawn(move || {
+            loop {
+                let score = match (src_bd, dst_bd) {
+                    (8, 8) => calc_score::<u8, u8>(&decoders, &src_config, &dst_config),
+                    (8, _) => calc_score::<u8, u16>(&decoders, &src_config, &dst_config),
+                    (_, 8) => calc_score::<u16, u8>(&decoders, &src_config, &dst_config),
+                    (_, _) => calc_score::<u16, u16>(&decoders, &src_config, &dst_config),
+                };
+
+                if let Some(result) = score {
+                    result_tx.send(result).unwrap();
                 } else {
+                    // no score = no more frames to read
                     break;
                 }
             }
-        };
-        let src_xyb = Xyb::try_from(src_rgb).unwrap();
-        let dst_xyb = Xyb::try_from(dst_rgb).unwrap();
-        let result =
-            compute_frame_ssimulacra2(src_xyb, dst_xyb).expect("Failed to calculate ssimulacra2");
-        if verbose {
-            println!("Frame {frame}: {result:.8}");
```

Comment:
Did this functionality disappear?

Reply:
Yes, I completely forgot about this. Added it back with the result ordering

```
...
[00:00:01] [12.21 fps] Frame 16
Frame 16: 93.59588478
[00:00:01] [15.97 fps] Frame 17
Frame 17: 93.46646854
[00:00:01] [16.67 fps] Frame 18
Frame 18: 93.50813436
[00:00:01] [13.54 fps] Frame 19
Frame 20: 93.67350370                  
...                                                         ...
```